tests/integration/test_types_integration.py

```diff
@@ -200,6 +201,119 @@ def test_digest(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .add_field(sql="DATE '02001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .execute()
+
+
+def test_time(trino_connection):
+    time_0 = datetime.time(1, 23, 45)
+    time_3 = datetime.time(1, 23, 45, 123000)
+    time_6 = datetime.time(1, 23, 45, 123456)
+
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS TIME)", python=None) \
+        .add_field(sql="CAST(null AS TIME(0))", python=None) \
+        .add_field(sql="CAST(null AS TIME(3))", python=None) \
+        .add_field(sql="CAST(null AS TIME(6))", python=None) \
+        .add_field(sql="CAST(null AS TIME(9))", python=None) \
+        .add_field(sql="CAST(null AS TIME(12))", python=None) \
+        .add_field(sql="CAST('01:23:45' AS TIME(0))", python=time_0) \
+        .add_field(sql="TIME '01:23:45.123'", python=time_3) \
+        .add_field(sql="CAST('01:23:45.123' AS TIME(3))", python=time_3) \
+        .add_field(sql="CAST('01:23:45.123456' AS TIME(6))", python=time_6) \
+        .add_field(sql="CAST('01:23:45.123456789' AS TIME(9))", python=time_6) \
+        .add_field(sql="CAST('01:23:45.123456789123' AS TIME(12))", python=time_6) \
+        .execute()
+
+
+def test_time_with_timezone(trino_connection):
+    query_time_with_timezone(trino_connection, '-08:00')
+    query_time_with_timezone(trino_connection, '+08:00')
+    query_time_with_timezone(trino_connection, '+05:30')
+
+
+def query_time_with_timezone(trino_connection, timezone_str):
+    tz = datetime.datetime.strptime(timezone_str, "%z").tzinfo
+    time_0 = datetime.time(1, 23, 45, 0).replace(tzinfo=tz)
+    time_3 = datetime.time(1, 23, 45, 123000).replace(tzinfo=tz)
+    time_6 = datetime.time(1, 23, 45, 123456).replace(tzinfo=tz)
+
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS TIME WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST(null AS TIME(0) WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST(null AS TIME(3) WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST(null AS TIME(6) WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST(null AS TIME(9) WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST(null AS TIME(12) WITH TIME ZONE)", python=None) \
+        .add_field(sql="CAST('01:23:45 %s' AS TIME(0) WITH TIME ZONE)" % (timezone_str), python=time_0) \
+        .add_field(sql="TIME '01:23:45.123 %s'" % (timezone_str), python=time_3) \
+        .add_field(sql="CAST('01:23:45.123 %s' AS TIME(3) WITH TIME ZONE)" % (timezone_str), python=time_3) \
+        .add_field(sql="CAST('01:23:45.123456 %s' AS TIME(6) WITH TIME ZONE)" % (timezone_str), python=time_6) \
+        .add_field(sql="CAST('01:23:45.123456789 %s' AS TIME(9) WITH TIME ZONE)" % (timezone_str), python=time_6) \
```

Comment:
That may be a naive question, but is there a way of reading results with data types with high precision without them getting truncated? Would it require modifying the query to cast them to some other data type or can the driver be configured to return them in some raw form, like strings?

Reply:
i don't think we should automatically switch between `datetime.time` and `str` depending on the precision.
in JDBC, user needs to explicitly ask for a String to get one (this works for all precisions).
Is it possible with Python cursor API?

Reply:
I didn't suggest switching based on precision. There must be an option for users that work with high-precision timestamps.

Reply:
@findepi Note that until recently everything returned by the Python client was untyped and `str`.

So it's still possible to return `str` when `experimental_python_types` is not enabled on the cursor.

But yes, for strongly typed time objects Python's stdlib is very limiting and there's nothing like `getObject(idx, clazz)` in DB-API.

Reply:
What other database clients do? (for DBs with more than ms precision)

Reply:
I know of only Oracle as another database that supports higher than millis precision and they don't expose that to the Python client since stdlib is limited to milliseconds.

Reply:
PostgreSQL, MySQL support microsecond timestamps
SQL Server, SAP HANA support timestamp with p=7

Reply:
Sorry I meant micros above. Micros is supported in Python stdlib so no issues with that. I'll see SQL Server and HANA. Thanks for the tip.

Reply:
@lpoulain If we're actually performing rounding then `01:23:45.123456789` should've been read back as `01:23:45.123457` instead.